### ISSUE TITLE
bug(sidenav): sidenav width "jumps" when a scrollbar is added

### ISSUE THREAD

**Issue by dario-piotrowicz** [angular/components#22965]
#### Reproduction

Use StackBlitz to reproduce your issue: 
   - https://stackblitz.com/edit/angular-unlegu?file=src%2Fapp%2Fsidenav-responsive-example.ts

Steps to reproduce:
1. Click on the hamburger icon to open the menu
2. Click on "Add" until a scrollbar appear
3. notice behaviour
4. Can be repeated by clicking on "Remove" until the scrollbar disappears and then "Add" until it reappears
 
 
#### Expected Behavior

When the scrollbar needs to appear, the navbar width should be increased so that it can fit the scrollbar without delay
(possibly the width increase should happen before the scrollbar appears)

#### Actual Behavior

When the scrollbar needs to appear, it first appears and then the navbar increases in width so that it can fit the scrollbar
(possibly the width increase should happen before the scrollbar appears)

(not the best user experience)

#### Screenshot

From stackblitz:
![Peek 2021-06-12 15-23](https://user-images.githubusercontent.com/61631103/121780526-b067eb00-cb98-11eb-9172-13ed442ef43c.gif)

From Angular.io/docs (at least I believe that it's the same bug):
![nav_last_issue](https://user-images.githubusercontent.com/61631103/121780562-d3929a80-cb98-11eb-8186-e52ea77d7f2e.gif)
(here you can see the chevron icons jumping when the scrollbar appears)
(I mentioned the issue here: https://github.com/angular/angular/issues/42552)

#### Environment

- Angular: 12
- CDK/Material: 12.0.4
- Browser(s): Version 91.0.4472.101
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu 20.04.2 LTS

**Comment by dario-piotrowicz** [comment]
Also I would like to mention that I've found a similar issue already present: https://github.com/angular/components/issues/22909

But I believe that this two are likely two different bugs

**Comment by lgaida** [comment]
Hi @dario-piotrowicz 
Have you tried using [autosize](https://material.angular.io/components/sidenav/overview#resizing-an-open-sidenav) on your sidenav-container?

**Comment by dario-piotrowicz** [comment]
Hi @lgaida I just noticed the stackblitz is no longer valid, I recreated it: https://stackblitz.com/edit/angular-4psyzl?file=src%2Fapp%2Fsidenav-responsive-example.html

and as you can see I added `autosize` but I don't think it's making any difference

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
